hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -1123,12 +1127,26 @@ rpcControllerFactory, getMetaLookupPool(), connectionConfig.getMetaReadRpcTimeou
     }
   }
 
-  void takeUserRegionLock() throws IOException {
+  private long heldStartTime;
```

Comment:
nit: this probably belongs at the top of the file with the other class variables. Since it is a class variable, maybe we can give it a more descriptive name to tie it to the lock
```suggestion
  private long userRegionLockHeldStartTime;
```

Reply:
Agree